From b08b61fa2b26fd08236ab7c010084901b05f2b04 Mon Sep 17 00:00:00 2001
From: zoucao <zoucaox@outlook.com>
Date: Thu, 8 Jun 2017 10:33:47 +0800
Subject: [PATCH 1/8] new_am335x: add new am335x base configs

Note: it just make new_am335x_defconfig, then generate configs, but
configs item has error

Signed-off-by: zoucao <zoucaox@outlook.com>
---
 arch/arm/Kconfig             |  8 ++++++++
 configs/new_am335x_defconfig |  3 +++
 include/configs/new_am335x.h | 17 +++++++++++++++++
 3 files changed, 28 insertions(+)
 create mode 100644 configs/new_am335x_defconfig
 create mode 100644 include/configs/new_am335x.h

diff --git a/arch/arm/Kconfig b/arch/arm/Kconfig
index 9908b43..67b535d 100644
--- a/arch/arm/Kconfig
+++ b/arch/arm/Kconfig
@@ -390,6 +390,13 @@ config TARGET_AM335X_EVM
 	select DM_SERIAL
 	select DM_GPIO
 
+config TARGET_NEWBOARD_AM335X
+	bool "Support newboard_am335x"
+	select CPU_V7
+	select DM
+	select DM_SERIAL
+	select DM_GPIO
+
 config TARGET_AM43XX_EVM
 	bool "Support am43xx_evm"
 	select CPU_V7
@@ -984,6 +991,7 @@ source "board/work-microwave/work_92105/Kconfig"
 source "board/xaeniax/Kconfig"
 source "board/xilinx/zynqmp/Kconfig"
 source "board/zipitz2/Kconfig"
+source "board/newboard_am335x/Kconfig"
 
 source "arch/arm/Kconfig.debug"
 
diff --git a/configs/new_am335x_defconfig b/configs/new_am335x_defconfig
new file mode 100644
index 0000000..c09d359
--- /dev/null
+++ b/configs/new_am335x_defconfig
@@ -0,0 +1,3 @@
+CONFIG_ARM=y
+CONFIG_TARGET_NEWBOARD_AM335X=y
+CONFIG_SPL=y
diff --git a/include/configs/new_am335x.h b/include/configs/new_am335x.h
new file mode 100644
index 0000000..1a682db
--- /dev/null
+++ b/include/configs/new_am335x.h
@@ -0,0 +1,17 @@
+#ifndef __CONFIG_NEWBAORD_AM335X_H
+#define __CONFIG_NEWBAORD_AM335X_H
+
+#ifndef CONFIG_SPL_BUILD
+#ifndef CONFIG_FIT
+# define CONFIG_FIT
+#endif
+# define CONFIG_TIMESTAMP
+# define CONFIG_LZO
+#endif
+
+/* Always 128 KiB env size */
+#define CONFIG_ENV_SIZE			(128 << 10)
+#define CONFIG_BAUDRATE			115200
+
+#define CONFIG_SYS_TEXT_BASE		0x08000000
+#endif
-- 
1.9.1

